wasm-commp/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+use std::io::{Cursor, Read};
+
+use primitives::{
+    commitment::{piece::PaddedPieceSize, CommP, Commitment},
+    NODE_SIZE,
+};
+use rs_merkle::MerkleTree;
+use wasm_bindgen::prelude::*;
+
+use crate::{fr32_reader::Fr32Reader, hasher::Sha256, zero_reader::ZeroPaddingReader};
+
+mod fr32_reader;
+mod hasher;
+mod zero_reader;
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(js_namespace = console)]
+    fn log(s: &str);
+}
```

Comment:
You could use tracing btw
https://github.com/eigerco/lumina/blob/main/node-wasm/src/utils.rs#L40-L52